src/code/audio/code_8019AF00.c

```diff
@@ -1953,17 +1957,152 @@ const char sAudioOcarinaUnusedText7[] = "check is over!!! %d %d %d\n";
+ * Will check to see that the notes are valid by ensuring no playable song is within the selected notes
+ * All OoT songs are valid, so the outer loop will run a maxiumum of two times.
+ * i.e. if random notes fails, then the next set of notes will be from a valid OoT song
+ */
+void AudioOcarina_TerminaWallGenerateNotes(void) {
+    OcarinaNote* ocarinaNote;
+    u8 randButton;
+    u8 i;
+    u8 j;
+
+    do {
+        i = 0;
+        if (sOcarinaWallCounter++ % 2) {
+            j = 0;
+
+            for (; i < 8; i++) {
+                randButton = Audio_NextRandom() % 5;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].pitch = sButtonToPitchMap[randButton];
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].length = 19;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].volume = 80;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].vibrato = 0;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].bend = 0;
+                j++;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].pitch = OCARINA_PITCH_NONE;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].length = 3;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].volume = 0;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].vibrato = 0;
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j].bend = 0;
+                j++;
+            }
+
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j - 2].length = 90;
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j - 1].length = 22;
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j + 1].pitch = OCARINA_PITCH_NONE;
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][j + 1].length = 0;
+            AudioOcarina_MapSongFromNotesToButtons(OCARINA_SONG_TERMINA_WALL, OCARINA_SONG_TERMINA_WALL, 8);
+        } else {
+            j = Audio_NextRandom() % 9;
+            ocarinaNote = sOoTOcarinaSongNotes[j];
+
+            for (; ocarinaNote[i].length != 0; i++) {
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][i] = ocarinaNote[i];
+            }
+
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][i].pitch = OCARINA_PITCH_NONE;
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][i].length = 0;
+            AudioOcarina_MapSongFromNotesToButtons(OCARINA_SONG_TERMINA_WALL, OCARINA_SONG_TERMINA_WALL,
+                                                   sOoTOcarinaSongsNumNotes[j]);
+        }
+    } while (AudioOcarina_TerminaWallValidateNotes(OCARINA_SONG_TERMINA_WALL, OCARINA_SONG_TERMINA_WALL) != 0);
+}
+
+/**
+ * Unused remnant of OoT
+ */
+void AudioOcarina_MemoryGameSetNumNotes(u8 minigameRound) {
+    u8 i;
+
+    if (minigameRound > 2) {
+        minigameRound = 2;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D26C.s")
+    sOcaMemoryGameAppendPos = 0;
```

Comment:
Why does some of this code use "oca" and some use "ocarina"? I'd rather it be consistent, and I'd much rather use "ocarina" throughout, since to me "oca" is not obviously an abbreviation of "ocarina".